Tests/FluentAssertions.Specs/Collections/Data/DataColumnCollectionAssertionExtensionsSpecs.cs

```diff
@@ -0,0 +1,433 @@
+        [Fact]
+        public void When_testing_that_references_to_the_same_object_are_the_same_it_should_succeed()
+        {
+            // Arrange
+            var dataTable = new DataTable("Test");
+
+            var columnCollection1 = dataTable.Columns;
+            var columnCollection2 = columnCollection1;
+
+            // Act & Assert
+            columnCollection1.Should().BeSameAs(columnCollection2);
+        }
+
+        [Fact]
+        public void When_testing_that_references_to_the_same_object_are_not_the_same_it_should_fail()
+        {
+            // Arrange
+            var dataTable = new DataTable("Test");
+
+            var columnCollection1 = dataTable.Columns;
+            var columnCollection2 = columnCollection1;
+
+            // Act
+            Action action =
+                () => columnCollection1.Should().NotBeSameAs(columnCollection2);
+
+            // Assert
+            action.Should().Throw<XunitException>();
+        }
+
+        [Fact]
+        public void When_testing_that_references_to_different_objects_are_the_same_it_should_fail()
+        {
+            // Arrange
+            var dataTable1 = new DataTable("Test1");
+            var dataTable2 = new DataTable("Test2");
+
+            var columnCollection1 = dataTable1.Columns;
+            var columnCollection2 = dataTable2.Columns;
+
+            // Act
+            Action action =
+                () => columnCollection1.Should().BeSameAs(columnCollection2);
+
+            // Assert
+            action.Should().Throw<XunitException>();
+        }
+
+        [Fact]
+        public void When_testing_that_references_to_different_objects_are_not_the_same_it_should_succeed()
+        {
+            // Arrange
+            var dataTable1 = new DataTable("Test1");
+            var dataTable2 = new DataTable("Test2");
+
+            var columnCollection1 = dataTable1.Columns;
+            var columnCollection2 = dataTable2.Columns;
+
+            // Act & Assert
+            columnCollection1.Should().NotBeSameAs(columnCollection2);
+        }
+        #endregion
+
+        #region HaveSameCount & NotHaveSameCount
+        [Fact]
+        public void When_asserting_same_count_if_expectation_is_null_it_should_fail()
+        {
+            // Arrange
+            var dataTable = new DataTable();
+
+            for (int seed = 0; seed < 3; seed++)
+            {
+                dataTable.Columns.Add(CreateTestDataColumn(seed));
+            }
+
+            var nullReference = default(DataColumnCollection);
+
+            // Act
+            Action action =
+                () => dataTable.Columns.Should().HaveSameCount(nullReference);
+
+            // Assert
+            action.Should().Throw<ArgumentNullException>().WithMessage(
+                "Cannot verify count against a <null> collection.*");
+        }
+
+        [Fact]
+        public void When_two_DataColumnCollections_have_the_same_number_elements_it_should_succeed()
```

Comment:
❌ Never use the literal name of a code element in the test name.

Reply:
In this case, the distinction is relevant because the implementation has different paths for generic collections and typed collections _of that specific type_. But, maybe the type name could be identified in a containing class...